### PR TITLE
More trace improvements

### PR DESCRIPTION
**Pull request by manasdk**
* Ability to skip noop triggers. This should help reduce noise

```
(virtualenv)manas@st2dev:~/st2$ st2 trace get 5668aea232ed353b0c46de89
id: 5668aea232ed353b0c46de89
trace_tag: trigger_instance-5668aea232ed353b0c46de88
start_timestamp: 2015-12-09T22:43:46.889177Z
+--------------------------+------------------+---------------------------+---------------------------+
| id                       | type             | ref                       | updated_at                |
+--------------------------+------------------+---------------------------+---------------------------+
| 5668aea232ed353b0c46de88 | trigger-instance | fixtures.test_passive_tri | 2015-12-09T22:43:46.88924 |
|                          |                  | gger.dummy                | 1Z                        |
| 5666295032ed355785437236 | rule             | default.passive_trigger_r | 2015-12-09T22:43:46.91876 |
|                          |                  | ule                       | 4Z                        |
| 5668aea232ed353b0c46de8b | execution        | core.local                | 2015-12-09T22:43:46.97451 |
|                          |                  |                           | 3Z                        |
| 5668aea332ed353b0c46de8d | trigger-instance | core.st2.generic.actiontr | 2015-12-09T22:43:47.15511 |
|                          |                  | igger                     | 1Z                        |
+--------------------------+------------------+---------------------------+---------------------------+
(virtualenv)manas@st2dev:~/st2$ st2 trace get 5668aea232ed353b0c46de89 --hide-noop-triggers
id: 5668aea232ed353b0c46de89
trace_tag: trigger_instance-5668aea232ed353b0c46de88
start_timestamp: 2015-12-09T22:43:46.889177Z
+--------------------------+------------------+---------------------------+---------------------------+
| id                       | type             | ref                       | updated_at                |
+--------------------------+------------------+---------------------------+---------------------------+
| 5668aea232ed353b0c46de88 | trigger-instance | fixtures.test_passive_tri | 2015-12-09T22:43:46.88924 |
|                          |                  | gger.dummy                | 1Z                        |
| 5666295032ed355785437236 | rule             | default.passive_trigger_r | 2015-12-09T22:43:46.91876 |
|                          |                  | ule                       | 4Z                        |
| 5668aea232ed353b0c46de8b | execution        | core.local                | 2015-12-09T22:43:46.97451 |
|                          |                  |                           | 3Z                        |
+--------------------------+------------------+---------------------------+---------------------------+
```

* Now ref of the most meaningful object is included so that the trace is easier to parse.

### CODE REVIEW

File: docs/source/traces.rst [hunk 1]
```diff
@@ -146,6 +146,24 @@ Get
 
     $ st2 trace get <trace-id>
 
+* View the causation chain in a trace for an action execution. Similarly for rule and trigger-instance.
```
Comment (Kami) [inline docs/source/traces.rst]: Would also be good to explain what causation (and causation) chain is in this context :trollface:

File: st2client/st2client/commands/trace.py [hunk 2]
```diff
@@ -299,7 +304,25 @@ def _apply_display_filters(trace, args):
...
+
+        # check if noop_triggers are to be hidden. This check applies whenever TriggerInstances
+        # are to be shown.
+        if (all_component_types or args.show_trigger_instances) and args.hide_noop_triggers:
+            filtered_trigger_instances = []
+            for trigger_instance in trace.trigger_instances:
+                is_noop_trigger_instance = True
+                for rule in trace.rules:
+                    caused_by_id = rule.get('caused_by', {}).get('id', None)
+                    if caused_by_id == trigger_instance['object_id']:
```
Comment (Kami) [inline st2client/st2client/commands/trace.py]: So deep :P

File: docs/source/traces.rst (hunk 1 above)
Comment (manasdk) [inline docs/source/traces.rst]: Yeah I need to work on the verbiage for sure. I don't have a succinct way of expressing this idea. I will take a stab at this once I get to using RuleEnforcement. Will likely have given some thought on how to express this better anyway by then.

File: st2common/st2common/models/db/trace.py [hunk 3]
```diff
@@ -31,6 +31,7 @@ class TraceComponentDB(me.EmbeddedDocument):
     """
     """
     object_id = me.StringField()
+    ref = me.StringField(default='')
```
Comment (Kami) [inline st2common/st2common/models/db/trace.py]: Hm, I don't think we default to an empty string anywhere else.

File: st2client/st2client/commands/trace.py (hunk 2 above)
Comment (manasdk) [inline st2client/st2client/commands/trace.py]: Yeah it is terrible. Given that we would not be dealing with rather large collections here I think it is an ok option.

File: st2common/st2common/models/db/trace.py (hunk 3 above)
Comment (manasdk) [inline st2common/st2common/models/db/trace.py]: shows up as `None/null` otherwise and it is mostly for backward compat anyway.

**Pull request merged**